src/app/vertextool/qgsvertexeditor.cpp

```diff
@@ -499,4 +499,11 @@ void CoordinateItemDelegate::setModelData( QWidget *editor, QAbstractItemModel *
   }
 }
 
-
+void CoordinateItemDelegate::setEditorData( QWidget *editor, const QModelIndex &index ) const
+{
+  QLineEdit *lineEdit = qobject_cast<QLineEdit *>( editor );
+  if ( lineEdit && index.isValid() )
+  {
+    lineEdit->setText( QLocale().toString( index.data( ).toDouble( ) ) );
```

Comment:
Should we use a fixed precision here to avoid scientific notation values?

Reply:
Yes, good point, I'll use the same format as in displayText.